runtime/bin/BUILD.gn

```diff
@@ -913,38 +913,35 @@ dart_executable("dart_precompiled_runtime_product") {
   extra_deps += [ ":elf_loader_product" ]
 }
 
-# This flag is set in runtime/runtime_args.gni
-# The analyze_snapshot tool is only supported on 64 bit AOT builds running under
-# linux and android platforms
-if (build_analyze_snapshot) {
```

Comment:
It might be easier to land this change w/o removing these guards (or just adding MacOS to the list).  I just removed them in our codebase as it seemed to work on all the targets we build, or has so far.